app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx

```diff
@@ -232,9 +238,29 @@ export function AuthProvider(props: AuthProviderProps) {
       }
     });
 
+  const setUsername = async (
+    accessToken: string,
+    username: string,
+    email: string
+  ) => {
+    const body: backendService.SetUsernameRequestBody = {
+      userName: username,
+      userEmail: email,
+    };
+
+    /** TODO [NP]: https://github.com/enso-org/cloud-v2/issues/343
+     * Don't create a new API client here, reuse the one from the context. */
```

Comment:
A TODO/FIXME in general should describe a problem, potential issues derived from that problem, potential solution and shouldd link to issue. Here we have solution and link, but there is no info about what problems the current solution causes. It is performance related?